toolkit/tools/internal/safemount/safemount_test.go

```diff
@@ -0,0 +1,124 @@
+	"github.com/moby/sys/mountinfo"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestResourceBusy(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Short mode enabled")
+	}
+
+	if !buildpipeline.IsRegularBuild() {
+		t.Skip("loopback block device not available")
+	}
+
+	if os.Geteuid() != 0 {
+		t.Skip("Test must be run as root because it uses loopback devices")
+	}
+
+	buildDir := filepath.Join(tmpDir, "TestResourceBusy")
+	err := os.MkdirAll(buildDir, 0o770)
+	if !assert.NoError(t, err, "failed to test temp directory (%s)", buildDir) {
+		return
+	}
+
+	diskConfig := configuration.Disk{
+		PartitionTableType: configuration.PartitionTableTypeGpt,
+		MaxSize:            4096,
+		Partitions: []configuration.Partition{
+			{
+				ID:     "a",
+				Start:  1,
+				End:    0,
+				FsType: "ext4",
+			},
+		},
+	}
+
+	// Create raw disk image file.
+	rawDisk, err := diskutils.CreateEmptyDisk(buildDir, "disk.raw", diskConfig.MaxSize)
+	assert.NoError(t, err, "failed to create empty disk file (%s)", buildDir)
+
+	// Connect raw disk image file.
+	loopback, err := safeloopback.NewLoopback(rawDisk)
+	if !assert.NoError(t, err, "failed to mount raw disk as a loopback device (%s)", rawDisk) {
+		return
+	}
+	defer loopback.Close()
+
+	// Set up partitions.
+	_, _, _, _, err = diskutils.CreatePartitions(loopback.DevicePath(), diskConfig,
+		configuration.RootEncryption{}, configuration.ReadOnlyVerityRoot{})
+	if !assert.NoError(t, err, "failed to create partitions on disk", loopback.DevicePath()) {
+		return
+	}
+
+	// Mount the partition.
+	partitionDevPath := loopback.DevicePath() + "p1"
+	partitionMountPath := filepath.Join(buildDir, "mount")
+
+	mount, err := NewMount(partitionDevPath, partitionMountPath, "ext4", 0, "", true)
+	if !assert.NoError(t, err, "failed to mount partition", partitionDevPath, partitionMountPath) {
+		return
+	}
+	defer mount.Close()
+
+	// Check that the mount exists.
+	exists, err := file.PathExists(partitionMountPath)
+	if !assert.NoError(t, err, "failed to check if mount directory exists") {
+		return
+	}
+	if !assert.Equal(t, true, exists, "mount directory doesn't exist") {
+		return
+	}
+
+	isMounted, err := mountinfo.Mounted(partitionMountPath)
+	if !assert.NoError(t, err, "failed to check if directory is not a mount point") {
+		return
+	}
+	if !assert.Equal(t, true, isMounted, "directory is not a mount point") {
+		return
+	}
+
+	// Open a file.
+	fileOnPartitionPath := filepath.Join(partitionMountPath, "test")
+
+	fileOnPartition, err := os.OpenFile(fileOnPartitionPath, os.O_RDWR|os.O_CREATE, 0)
+	if !assert.NoErrorf(t, err, "failed to open file", fileOnPartitionPath) {
+		return
+	}
+	defer fileOnPartition.Close()
+
+	// Try to close the mount.
+	err = mount.CleanClose()
+	assert.Error(t, err)
+	assert.ErrorContains(t, err, "busy")
+
+	// Close the file.
+	fileOnPartition.Close()
+
+	// Try to close the mount again.
+	err = mount.CleanClose()
+	assert.NoError(t, err, "failed to close the mount")
```

Comment:
To better exercise the retry, maybe something like this?
I did some similar patterns in the retry unit tests

```suggestion
	// Close the file.
	go func() {
		time.Sleep(time.Duration(2)  * time.Second)
		fileOnPartition.Close()
	}

	// Try to close the mount again.
	time.Sleep(time.Duration(1) * time.Second)
	err = mount.CleanClose()
	assert.NoError(t, err, "failed to close the mount")
```

Reply:
That might be a little prone to race conditions. However, I did add a duration check for the first close() to sanity check that the retries are actually occurring.